src/widgets/DiagramWidget.tsx

```diff
@@ -299,13 +299,39 @@ export class DiagramWidget extends React.Component<DiagramProps, DiagramState> {
 				if (!(model.model instanceof PointModel)) {
 					return;
 				}
-
 				if (element && element.model instanceof PortModel && !diagramEngine.isModelLocked(element.model)) {
 					linkConnected = true;
 					let link = model.model.getLink();
-					link.setTargetPort(element.model);
+					if(link.getTargetPort() !== null) 
+					{
```

Comment:
I think just run this code through the lintjs script defined in the package.json so we can keep the brackets on the same line etc..